src/pages/permissions/CreateIdentityModal.tsx

```diff
@@ -0,0 +1,98 @@
+import { FC, useState } from "react";
+import { ActionButton, Input, Modal } from "@canonical/react-components";
+import { useFormik } from "formik";
+import Loader from "components/Loader";
+import { fetchIdentityDetails } from "api/auth-identities";
+import { base64EncodeTokenDetails } from "util/helpers";
+
+interface Props {
+  onClose: () => void;
+}
+
+const CreateIdentityModal: FC<Props> = ({ onClose }) => {
+  const [isLoadingToken, setIsLoadingToken] = useState(false);
+  const [token, setToken] = useState<string | null>(null);
+
+  const handleClose = () => {
+    onClose();
+    window.location.reload();
+  };
+
+  const formik = useFormik({
+    initialValues: {
+      clientName: "",
+    },
+    onSubmit: (values) => {
+      setIsLoadingToken(true);
+
+      fetchIdentityDetails(values.clientName)
+        .then((tokenPreResponse) => {
+          const tempToken = base64EncodeTokenDetails(tokenPreResponse);
+
+          setIsLoadingToken(false);
+          setToken(tempToken);
+        })
+        .catch((err) => {
+          setIsLoadingToken(false);
+          console.error("Error fetching token:", err);
+        });
+    },
+  });
+
+  return (
+    <Modal
+      close={onClose}
+      className="create-tls-identity"
+      title="Generate Trust Token"
+      buttonRow={
+        <>
+          {token ? (
+            <ActionButton
```

Comment:
We don't need a refresh button to update identities list, instead we should use react query invalidation